pkg/util/binding.go

```diff
@@ -76,16 +76,33 @@ func DivideReplicasByTargetCluster(clusters []workv1alpha2.TargetCluster, sum in
 		}
 		allocatedReplicas += res[i].Replicas
 	}
-	if remainReplicas := sum - allocatedReplicas; remainReplicas > 0 {
-		for i := 0; remainReplicas > 0; i++ {
-			if i == len(res) {
-				i = 0
-			}
+
+	divideRemainingReplicas(int(sum-allocatedReplicas), res)
+
+	return res
+}
+
+// divideRemainingReplicas divide remaining Replicas to clusters
+func divideRemainingReplicas(remainingReplicas int, res []workv1alpha2.TargetCluster) {
+	if remainingReplicas <= 0 {
+		return
+	}
+
+	clusterSize := len(res)
+	if remainingReplicas < clusterSize {
```

Comment:
When the value of `remainingReplicas` is several times the value of `clusterSize`, some calculations can be omitted.

Maybe we can remove judgment. If match the case `remainingReplicas < clusterSize`, the `avg` value is 0.